### PR TITLE
[GH-#396] Fix incorrect function specifications

### LINKED ISSUE #396: Fix incorrect function specifications

### PR DESCRIPTION
**Pull request by DanielaIvanova**
Resolves #396

### CODE REVIEW

File: apps/aecore/lib/aecore/account/tx/spend_tx.ex [hunk 1]
```diff
@@ -115,10 +117,10 @@ defmodule Aecore.Account.Tx.SpendTx do
   before the transaction is executed.
   """
   @spec preprocess_check(
-          ChainState.account(),
+          Chainstate.accounts(),
           tx_type_state(),
           non_neg_integer(),
-          SpendTx.t(),
+          t(),
```
Comment (thepiwo) [inline apps/aecore/lib/aecore/account/tx/spend_tx.ex]: I think the full spec is nicer here

Comment (DanielaIvanova) [reply apps/aecore/lib/aecore/account/tx/spend_tx.ex]: For Elixir it is custom to write the types without the module name (If we are in the current module of course). @rdimitrovbg and I looked into some of the Elixir documentations and this is how they do it.

Comment (thepiwo) [reply apps/aecore/lib/aecore/account/tx/spend_tx.ex]: It is my personal opinion, If I just read some code I can understand way better what is meant, whats your opinion @d-velev @gspasov @cytadela8 ?

Comment (gspasov) [reply apps/aecore/lib/aecore/account/tx/spend_tx.ex]: If this is the way Elixir does it, it should be the way to go.

Comment (thepiwo) [reply apps/aecore/lib/aecore/account/tx/spend_tx.ex]: What is your personal opinion?

Comment (gspasov) [reply apps/aecore/lib/aecore/account/tx/spend_tx.ex]: To leave it like this.

Comment (thepiwo) [reply apps/aecore/lib/aecore/account/tx/spend_tx.ex]: @d-velev @meivantodorov @cytadela8 (@untra maybe?)

**Pull request merged**